src/Domain/Model/PaymentReferenceCode.php

```diff
@@ -23,19 +23,24 @@ class PaymentReferenceCode {
 	public const LENGTH_CHECKSUM = 1;
 	public const READABILITY_DELIMITER = '-';
 
+	private ?int $id = null;
+	private string $formattedCode;
+
 	/**
 	 * @param string $prefix The accounting department uses the prefix to identify different types of payments without needing a lookup for each payment
 	 * @param string $code A unique code, using the allowed character set
 	 * @param string $checksum A Checksum character, derived from the concatenated prefix and code, with the number converted to the allowed characters set (@see ChecksumGenerator)
 	 */
 	public function __construct(
-		private string $prefix,
-		private string $code,
-		private string $checksum
+		private readonly string $prefix,
+		private readonly string $code,
+		private readonly string $checksum
```

Comment:
With Doctrine storing this as a formatted string and the column type being "string", the dehydrated payment reference code will be broken, since prefix, code and checksum won't be initialized. That's not a problem because we don't have methods any more that access them. My suggestion would be to remove the `private readonly` modifiers, since we don't need to access the individual parts.